src/kata6.joinNames.js

```diff
@@ -1,3 +1,16 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+  let returnStr = '';
+  let lastIndex = namesObj.length - 1;
+  namesObj.forEach( (item, index) => {
+
+    if (index === lastIndex) {
+      returnStr += ' & ';
+    } else if (index > 0) {
+      returnStr += ', ';
+    }
+    returnStr += item.name;
+  });
+  return returnStr;
```

Comment:
So theres a few things we could improve here:
- `lastIndex` is never reassigned, so the appropriate keyword to use would be `const` and not `let`
- on line 4 there is an extra space before the brackets containing `item` and `index`, should be taken out
- Quite often its a convention have a clear line above the final return statement in a function. This isnt strictly necessary but have added it here so youre aware of it.
All together this should like this:
```
  let returnStr = "";
  const lastIndex = namesObj.length - 1;

  namesObj.forEach((item, index) => {
    if (index === lastIndex) {
      returnStr += " & ";
    } else if (index > 0) {
      returnStr += ", ";
    }
    returnStr += item.name;
  });

  return returnStr;
```